sycl/source/detail/scheduler/commands.cpp

```diff
@@ -1481,9 +1521,40 @@ void DispatchNativeKernel(void *Blob) {
   HostTask->MHostKernel->call(HostTask->MNDRDesc, nullptr);
 }
 
+struct HostTaskContext {
+  CGHostTask *HostTask;
+
+  // events dependencies
+  std::map<const detail::plugin *, std::vector<EventImplPtr>>
+      RequiredEventsPerPlugin;
+
+  // Context with which SelfEvent has to be created
+  ContextImplPtr Context;
+
+  EventImplPtr SelfEvent;
+};
+
+void DispatchHostTask(const std::shared_ptr<HostTaskContext> &Ctx) {
+  // wait for dependency events
+  // FIXME introduce a more sophisticated wait mechanism
+  for (auto &PluginWithEvents : Ctx->RequiredEventsPerPlugin) {
+    auto RawEvents = getPiEvents(PluginWithEvents.second);
+    PluginWithEvents.first->call<PiApiKind::piEventsWait>(RawEvents.size(),
+                                                          RawEvents.data());
+  }
+
+  // we're ready to call the user-defined lambda now
+  Ctx->HostTask->MHostTask->call();
+
+  const detail::plugin &Plugin = Ctx->SelfEvent->getPlugin();
+  Plugin.call<PiApiKind::piEventSetStatus>(Ctx->SelfEvent->getHandleRef(),
```

Comment:
We cannot create event in this way when the queue(context) is a host one.

Reply:
Done.